crates/oxc_regular_expression/src/parser/reader/string_literal_parser/mod.rs

```diff
@@ -240,4 +243,14 @@ without continuation""#,
             assert_eq!(a.value, b.value);
         }
     }
+
+    #[test]
+    fn regexp_literal_with_template_literal_and_new_line() {
+        let source_text = r"re = new RegExp(`^12
+        34$`)";
+
+        let result = Parser::new(source_text, Options::default()).parse();
+
+        assert!(result.is_ok(), "Expected to parse: {source_text} but failed: {:?}", result.err());
```

Comment:
@leaysgur I do not have deep knowledge about the regex parser. 

Line breaks in template literals are allowed with regex. I want to cover the test case from `no-misleading-character-class`:
https://github.com/eslint/eslint/blob/0c9999c2a682151cf13bb3a4f8916930678c2f9b/tests/lib/rules/no-misleading-character-class.js#L841-L858

Can you guide me into the right direction or contribute to the PR :)?
PS: The naming is now for `StringLiteral(Kind)` is now wrong. Happy for name suggestions^^"